www/apps/docs/content/deployments/storefront/deploying-next-on-aws-using-terraform.md

```diff
@@ -0,0 +1,170 @@
+
+  ```shell
+  aws sts get-caller-identity \
+      --query Account \
+      --output text
+  ```
+
+If you created your ECR repository using the [Terraform Module for Medusa on AWS](https://registry.terraform.io/modules/u11d-com/medusajs/aws), you can find the ECR repository URL in the module's output values after deployment.
+
+### Step 2: Tag the Docker Container Image
+
+Tag the Docker container image with your ECR repository URI:
+
+```shell
+docker tag medusa_storefront:1.0.0 <aws_account_id>.dkr.ecr.<region>.amazonaws.com/<repository>:<tag>
+```
+
+Replace:
+
+- `medusa_storefront:1.0.0` with the image tag you created earlier.
+- `<aws_account_id>`, `<region>`, `<repository>`, and `<tag>` with your AWS account ID, region, ECR repository name, and desired image tag, respectively.
+
+### Step 3: Push the Docker Image
+
+Push the Docker container image to your ECR repository:
+
+```shell
+docker push <aws_account_id>.dkr.ecr.<region>.amazonaws.com/<repository>:<tag>
+```
+
+Example:
+
+```shell
+docker push 123456789012.dkr.ecr.us-east-1.amazonaws.com/my-storefront:1.0.0
+```
+
+For more details, refer to the [AWS ECR documentation](https://docs.aws.amazon.com/AmazonECR/latest/userguide/docker-push-ecr-image.html).
+
+---
+
+## Connecting the Storefront to Your Terraform Infrastructure
+
+After pushing the Docker image, connect it to your infrastructure deployed via Terraform:
+
+1. Reference the ECR Image in Terraform:
+Add the ECR repository URI and tag to your Terraform configuration:
+
+```hcl
+storefront_container_image = "<aws_account_id>.dkr.ecr.<region>.amazonaws.com/<repository>:<tag>"
+```
+
+2. Enable storefront deployment:
+Ensure the storefront is enabled in your Terraform module configuration:
+
+```hcl
+storefront_create = true
+```
+
+3. Plan and apply Terraform changes:
+Deploy the updated configuration:
+
+```shell
+terraform init
+terraform plan
+terraform apply
+```
+
+Confirm the changes when prompted.
+
+---
+
+## Troubleshooting
+
+### Common Issues
+
+- **Docker Build Fails**:
+  - Ensure all dependencies are installed.
+  - Verify the `Dockerfile` is correctly configured.
+  - Check for syntax errors or missing files.
+- **ECR Authentication Fails**:
+  - Confirm that your AWS credentials are correctly configured with sufficient permissions.
+  - Ensure your AWS CLI session is active (re-authenticate if necessary).
+- **Image Push Fails**:
+  - Ensure the ECR repository exists.
+  - Ensure the repository URI and image tag are correctly specified.
+  - Check for network connectivity issues.
+- **Terraform Apply Fails**:
+  - Confirm that the referenced Docker image exists in the ECR repository.
+  - Check Terraform logs for specific error messages.
+
+## Additional Resources
+
+- [Medusa Documentation](https://docs.medusajs.com/v1/)
+- [Next.js Documentation](https://nextjs.org/)
+- [Terraform Module for Medusa on AWS Provider Documentation](https://registry.terraform.io/modules/u11d-com/medusajs/aws)
```

Comment:
Those links aren't really necessary. Maybe you can just link to the Terraform Module's documentation.